hack/schema_validation.sh

```diff
@@ -1,28 +1,115 @@
 #!/bin/bash
+###############################################################################
+#                                                                             #
+#      Check Managed Scripts metadata.yaml files against the JSON Schema      #
+#                                                                             #
+###############################################################################
+#
+# release: v2.0
+# description: Script responsible for listing the metadata.yaml files in the
+#              managed scripts directory and for running the validation 
+#              against the Json Schema.
+# team: SREP
+# author: Fabio Aldana
+# email: faldana@redhat.com
+#
 
 #set -ex
 
-cd $(dirname $0)/..
+# define variables
 CONTAINER_ENGINE=$(which docker 2>/dev/null || which podman)
-#convert all metadata.yaml to json first
-yfiles=$(find . -name 'metadata.yaml')
-for f in $yfiles
+CONTAINER_PATH=/json
+
+# change to the managed-scripts directory on your repository
+cd $(dirname $0)/..
+
+# funtion to present the help
+usage() {
+cat <<EOF >&2
+Usage: $0 <options> [script1] [scriptN]
+
+  Check Managed Scripts metadata.yaml files against the JSON Schema.
+
+  The script can be run with no arguments, which runs against all files found 
+  in the managed-scripts directory, or it can be specified the scripts to be
+  validated.
+
+  Options:
+
+    -h, --help        Shows the script usage.
+
+EOF
+  exit -1
+}
+
+# funtion which validates the list of scripts and adds the container path to each file found
+yamlList() {
+
+if [ ! -z "$listYamlFiles" ]; then
+  for i in $listYamlFiles
+  do
+    yamlFiles+="$CONTAINER_PATH$i "
+  done
+  # calls check_validation function to run the validation on the scripts
+  check_validation
+else
+  echo "error: no file was found."
+  exit 1
+fi
+
+}
+
+# funtion to validate the Yaml files
+check_validation() {
+
+echo "validating the jsonschema for $yamlFiles"
+$CONTAINER_ENGINE run --rm -v $(pwd):$CONTAINER_PATH quay.io/app-sre/managed-scripts:latest /root/.local/bin/check-jsonschema --schemafile $CONTAINER_PATH/hack/metadata.schema.json $yamlFiles
+
+}
+
+# funtion that list all files in the manage-scripts directory
+all() {
+
+# list all yaml files
+listYamlFiles="$(find . -name 'metadata.yaml' | cut -c 2- ) "
+
+# calls yamList function to prepare the scripts to be validated
+yamlList
+
+}
+
+# funtion that list the scripts passed as arguments
+partial() {
+
+# list the number of arguments
+for ((i = 1; i <= $#; i++ ));
 do
-   echo "convert $f to json"
-   if ! python3 hack/yamltojson.py $f; then
```

Comment:
if the convert is not needed anymore, should we remove the file `hack/yamltojson.py` as well?

Reply:
The file should not be there.

Reply:
Actually, they were removed already. They are in red.

Reply:
I mean the file itself in the same directory.

```
$ ls hack/ -l
total 20
-rwxr-xr-x. 1 bmeng bmeng   60 Jun 12 10:35 app_sre_build_deploy.sh
-rwxr-xr-x. 1 bmeng bmeng   81 Jun 12 10:35 app_sre_pr_check.sh
-rw-r--r--. 1 bmeng bmeng 3840 Sep 18 15:32 metadata.schema.json
-rwxr-xr-x. 1 bmeng bmeng 2972 Sep 18 16:05 schema_validation.sh
-rwxr-xr-x. 1 bmeng bmeng  227 Jun 12 10:35 yamltojson.py          <---------------------

Reply:
Yeah, got it, sorry.
Done, file removed.